tests/trainer/test_fsdp_checkpoint.py

```diff
@@ -461,13 +467,76 @@ def test_fsdp_load_old_checkpoint(
     state_dict2 = trainer.state.state_dict()
 
     if (dist.get_global_rank() == 0 and state_dict_type == 'full') or state_dict_type == 'sharded':
-        filled_load_path = load_path.format(rank=dist.get_global_rank())
-        destination = str(tmp_path / pathlib.Path(filled_load_path).name)
-        get_file(filled_load_path, destination=destination)
-        with open(destination, 'rb') as f:
-            state_dict1 = torch.load(f)['state']
-        _compare_model_params_between_state_dicts(state_dict1, state_dict2)
+        if state_dict_type == 'sharded' and composer_version >= '0.16.0':
```

Comment:
Should we use version package instead of strings? e.g. https://github.com/mosaicml/composer/blob/dev/composer/trainer/mosaic_fsdp.py

Reply:
fixed